Tools/SandboxTest.ps1

```diff
@@ -273,6 +305,132 @@ function Test-FileChecksum {
+                return $true
+            }
+            # If the parsing failed, the expiration should still be at the minimum value
+            elseif ($cachedExpirationDate -eq [System.DateTime]::MinValue) {
+                Write-Verbose 'The cached token contained content, but it could not be parsed as a date. It will be re-validated'
+                Invoke-FileCleanup -FilePaths $cachedToken.FullName
+                # Do not return anything, since the token will need to be re-validated
+            }
+            else {
+                Write-Verbose "The cached token contained content, but the token expired $([Math]::Abs($tokenExpirationDays)) days ago"
+                # Leave the cached token so that it doesn't throw script exceptions in the future
+                # Invoke-FileCleanup -FilePaths $cachedToken.FullName
+                return $false
+            }
+        }
+        else {
+            # Either the token was empty, or the cached token is expired. Remove the cached token so that re-validation
+            # of the token will update the date the token was cached if it is still valid
+            Invoke-FileCleanup -FilePaths $cachedToken.FullName
+        }
+    }
+    else {
+        Write-Verbose 'Token was not found in the cache'
+    }
+
+    # To get here either the token was not in the cache or it needs to be re-validated
+
+    $requestParameters = @{
+        Uri            = 'https://api.github.com/rate_limit'
+        Authentication = 'Bearer'
+        Token          = $(ConvertTo-SecureString "$Token" -AsPlainText)
+    }
+
+    Write-Verbose "Checking Token against $($requestParameters.Uri)"
+    $apiResponse = Invoke-WebRequest @requestParameters # This will return an exception if the token is not valid; It is intentionally not caught
+    # The headers can sometimes be a single string, or an array of strings. Cast them into an array anyways just for safety
+    $rateLimit = @($apiResponse.Headers['X-RateLimit-Limit'])
+    $tokenExpiration = @($apiResponse.Headers['github-authentication-token-expiration']) # This could be null if the token is set to never expire
+    Write-Debug "API responded with Rate Limit ($rateLimit) and Expiration ($tokenExpiration)"
+
+    if (!$rateLimit) { return $false } # Something went horribly wrong, and the rate limit isn't known. Assume the token is not valid
+    if ([int]$rateLimit[0] -le 60) {
+        # Authenticated users typically have a limit that is much higher than 60
+        return $false
+    }
+
+    Write-Verbose 'Token validated successfully. Adding to cache'
+    # If the token doesn't expire, write a special value to the file
+    if (!$tokenExpiration -or [string]::IsNullOrWhiteSpace($tokenExpiration[0])) { $tokenExpiration = @([System.DateTime]::MaxValue) }
+    # When datetime objects are converted to string, they have extra whitespace that should be trimmed off
+    $tokenExpiration = $tokenExpiration[0].ToLongDateString().Trim()
```

Comment:
```suggestion
    $tokenExpiration = [DateTime]::Parse($tokenExpiration[0]).ToLongDateString().Trim()
```

I'm really not sure why you and I get different GitHub API responses, but `$tokenExpiration[0]` is a string for me.


![image](https://github.com/user-attachments/assets/ddfcd210-9f3a-40fc-be6c-41037b907f31)